### PR TITLE
Use .reverse() on mutable slices.

### PR DESCRIPTION
**Pull request by Dr-Emann**

### CODE REVIEW

File: reversewords/src/lib.rs [hunk 1]
```diff
@@ -44,7 +38,7 @@ pub fn ascii_reverse_words(s: &mut String) {
         let ref mut bytes = s.as_mut_vec();  // Unsafe.
         // Reverse the entire string. So:
         // "Hello from Rust!" => "!tsuR morf olleH"
-        reverse_vec(bytes, 0, len-1);
+        reverse_vec(bytes, 0, len);
```
Comment (brianquinlan) [inline reversewords/src/lib.rs]: Can you inline bytes[0..len].reverse() here?

File: reversewords/src/lib.rs [hunk 2]
```diff
@@ -57,7 +51,7 @@ pub fn ascii_reverse_words(s: &mut String) {
                 while right < len && bytes[right] != SPACE {
                     right += 1;
                 }
-                reverse_vec(bytes, left, right - 1);
+                reverse_vec(bytes, left, right);
```
Comment (brianquinlan) [inline reversewords/src/lib.rs]: Can you inline bytes[left..right].reverse() here?

File: pancake/src/lib.rs [hunk 3]
```diff
@@ -1,8 +1,7 @@
 /// Reverse the items in the list from [0..index].
 #[inline]
 fn reverse_until_index<T>(v: &mut Vec<T>, index: usize) {
-    let (mut first, _) = v.split_at_mut(index+1);
-    first.reverse();
+    v[..index+1].reverse();
```
Comment (brianquinlan) [inline pancake/src/lib.rs:4]: Nice!

**Comment by Dr-Emann** [comment]
Rebased onto master

**Comment by brianquinlan** [comment]
Awesome, thanks!

**Pull request merged**